### PR TITLE
Fix libhdf5.so not found error

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Fix h5py so not found error

**Ticket:** TBD

### CODE REVIEW

File: src/bindings/python/constraints.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ wheel>=0.38.1
 patchelf<=0.17.2.1
 
 # Frontends
-h5py>=3.1.0,<3.12.0
+h5py>=3.1.0,<3.11.0
```
Comment (rkazants) [inline src/bindings/python/constraints.txt:19]:
```suggestion
h5py>=3.1.0,<3.12.0
```

File: src/frontends/tensorflow/tests/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
 -c ../../../bindings/python/constraints.txt
-numpy
 h5py
+numpy
```
Comment (rkazants) [inline src/frontends/tensorflow/tests/requirements.txt:3]:
```suggestion
numpy
h5py
```

File: src/frontends/tensorflow_lite/tests/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,4 @@
 -c ../../../bindings/python/constraints.txt
+h5py
```
Comment (rkazants) [inline src/frontends/tensorflow_lite/tests/requirements.txt:2]:
```suggestion
```

File: tests/layer_tests/requirements.txt [hunk 4]
```diff
@@ -1,5 +1,6 @@
 -c ../constraints.txt
 # paddlepaddle #  ticket 95904
+h5py
```
Comment (rkazants) [inline tests/layer_tests/requirements.txt:3]:
```suggestion
```

File: tests/model_hub_tests/performance_tests/requirements.txt [hunk 5]
```diff
@@ -1,4 +1,5 @@
 -c ../../constraints.txt
+h5py
```
Comment (rkazants) [inline tests/model_hub_tests/performance_tests/requirements.txt:2]:
```suggestion
```

File: tests/model_hub_tests/tensorflow/requirements.txt [hunk 6]
```diff
@@ -1,4 +1,5 @@
 -c ../../constraints.txt
+h5py
```
Comment (rkazants) [inline tests/model_hub_tests/tensorflow/requirements.txt:2]:
```suggestion
```

File: .github/workflows/job_tensorflow_layer_tests.yml [hunk 7]
```diff
@@ -116,6 +116,12 @@ jobs:
       # Tests
       #
 
+      - name: Install dependencies
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          # libhdf5-dev is required for proper h5py and tensorflow installation
+          apt update && apt install -y libhdf5-dev
+
```
Comment (rkazants) [inline .github/workflows/job_tensorflow_layer_tests.yml:124]:
```suggestion
```

File: .github/workflows/job_tensorflow_models_tests.yml [hunk 8]
```diff
@@ -99,7 +99,8 @@ jobs:
         if: ${{ runner.os == 'Linux' }}
         run: |
           # install git (required to build pip deps from the sources)
-          sudo apt-get install --assume-yes --no-install-recommends g++ git ca-certificates wget
+          # libhdf5-dev is required for proper h5py and tensorflow installation
+          sudo apt-get install --assume-yes --no-install-recommends g++ git ca-certificates wget libhdf5-dev
```
Comment (rkazants) [inline .github/workflows/job_tensorflow_models_tests.yml:103]:
```suggestion
          sudo apt-get install --assume-yes --no-install-recommends g++ git ca-certificates wget
```

File: .github/workflows/linux_arm64.yml [hunk 9]
```diff
@@ -130,6 +130,12 @@ jobs:
       # Dependencies
       #
 
+      - name: Install dependencies
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          # libhdf5-dev is required for proper h5py and tensorflow installation
+          apt update && apt install -y libhdf5-dev
+
```
Comment (rkazants) [inline .github/workflows/linux_arm64.yml:138]:
```suggestion
```

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by wkobielx** [comment]
Hi @rkazants we have the same issue on releases/2024/2 branch, are you going to port this PR? 
https://github.com/openvinotoolkit/openvino/actions/runs/9237833510/job/25415278666

**Comment by rkazants** [comment]
> Hi @rkazants we have the same issue on releases/2024/2 branch, are you going to port this PR? https://github.com/openvinotoolkit/openvino/actions/runs/9237833510/job/25415278666

https://github.com/openvinotoolkit/openvino/pull/24694

**Pull request merged**